code/processes/wdb.q

```diff
@@ -151,26 +151,37 @@ savetables:$[writedownmode~`partbyattr;savetablesbypart;savetables];
 
 savetodisk:{[] savetables[savedir;getpartition[];0b;] each tablelist[]};
 
-/- eod - flush remaining data to disk
 endofday:{[pt;processdata]
-	.lg.o[`eod;"end of day message received - ",spt:string pt];
+        .lg.o[`eod;"end of day message received - ",spt:string pt];
         /- set what type of merge method to be used
         mergemethod:.wdb.mergemode;
-	/- create a dictionary of tables and merge limits, byte or row count limit depending on settings
-	.lg.o[`merge;"merging partitons by ",$[.merge.mergebybytelimit;"byte estimate";"row count"]," limit"];
-	mergelimits:(tablelist[],())!($[.merge.mergebybytelimit;{(count x)#mergenumbytes};{[x] mergenumrows^mergemaxrows[x]}]tablelist[]),();	
-	tablist:tablelist[]!{0#value x} each tablelist[];
-	/ - if save mode is enabled then flush all data to disk
-	if[saveenabled;
-		endofdaysave[savedir;pt];
-		/ - if sort mode enable call endofdaysort within the process,else inform the sort and reload process to do it
-		$[sortenabled;endofdaysort;informsortandreload] . (savedir;pt;tablist;writedownmode;mergelimits;hdbsettings;mergemethod)];
-	.lg.o[`eod;"deleting data from ",$[r:writedownmode~`partbyattr;"partsizes";"tabsizes"]];
-	$[r;@[`.merge;`partsizes;0#];@[`.wdb;`tabsizes;0#]];
-	.lg.o[`eod;"end of day is now complete"];
-	.wdb.currentpartition:pt+1;
-	};
-	
+        /- create a dictionary of tables and merge limits, byte or row count limit depending on settings
+        .lg.o[`merge;"merging partitons by ",$[.merge.mergebybytelimit;"byte estimate";"row count"]," limit"];
+        mergelimits:(tablelist[],())!($[.merge.mergebybytelimit;{(count x)#mergenumbytes};{[x] mergenumrows^mergemaxrows[x]}]tablelist[]),();
+        tablist:tablelist[]!{0#value x} each tablelist[];
+        / Need to download sym file to scratch directory if this is Finspace application
+        if[.finspace.enabled;
+                        .lg.o[`createchangeset;"downloading sym file to scratch directory for ",.finspace.database];
+                        .aws.get_latest_sym_file[.finspace.database;getenv[`KDBSCRATCH]];
+        ];
```

Comment:
do we need to do this if save isn't enabled? (i.e. perhaps this should be inside the `if[saveenabled;` below, or maybe even inside the `endofdaysave` function?)

Reply:
Suppose yeah if save isn't enabled then there shouldn't be a changeset created/any saving done.